### PR TITLE
Request Tracker Cookbook

### PR DESCRIPTION
**Pull request by zwall-bp**
Under construction...

### CODE REVIEW

File: Berksfile [hunk 1]
```diff
@@ -2,7 +2,8 @@ source 'https://supermarket.chef.io'
 
 solver :ruby, :required
 
-cookbook 'osl-apache', git: 'git@github.com:osuosl-cookbooks/osl-apache'
+# cookbook 'osl-apache', git: 'git@github.com:osuosl-cookbooks/osl-apache'
+cookbook 'osl-apache', path: '../osl-apache'
```
Comment (ramereth) [inline Berksfile]: You'll need to push whatever changes you have to a branch and then update this so I can test it myself.

**Comment by zwall-bp** [comment]
For testing the suite, please use the [`zwall/rt-patches`](https://github.com/osuosl-cookbooks/osl-apache/pull/175) branch of `osl-apache`. I have configured the PR to assume the master branch has the patches applied.

**Comment by ramereth** [comment]
> For testing the suite, please use the [`zwall/rt-patches`](https://github.com/osuosl-cookbooks/osl-apache/pull/175) branch of `osl-apache`. I have configured the PR to assume the master branch has the patches applied.

Please go ahead and just have the Berksfile point to that branch since this is still being worked on.

File: attributes/default.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+default['osl-postfix']['main']['home_mailbox'] = 'Mail/'
+default['osl-postfix']['main']['mailbox_command'] = '/usr/bin/procmail'
+default['osl-postfix']['main']['mailbox_size_limit'] = '0'
+default['osl-postfix']['main']['message_size_limit'] = '102400000'
+default['osl-postfix']['main']['transport_maps'] = 'hash:/etc/postfix/transport'
+
+default['postfix']['access']['140.211.166.133'] = 'OK' # smtp2.osuosl.org
+default['postfix']['access']['140.211.166.136'] = 'OK' # smtp3.osuosl.org
+default['postfix']['access']['140.211.166.137'] = 'OK' # smtp4.osuosl.org
+default['postfix']['access']['140.211.166.138'] = 'OK' # smtp1.osuosl.org
```
Comment (ramereth) [inline attributes/default.rb]: These should all be moved into the recipe at the top with `node.default`

File: compliance/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+```sh
+# Generate a Cinc Auditor profile
+cinc generate profile PROFILE_NAME
+
+# Generate a Cinc Auditor waiver file
+cinc generate waiver WAIVER_NAME
+
+# Generate a Cinc Auditor input file
+cinc generate input INPUT_NAME
+```
```
Comment (ramereth) [inline compliance/README.md]: Please remove this file.

File: documentation/resource_osl_request_tracker.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+```rb
+default['osl-rt']['queues'].tap |q| do
+  q['Pretty Text Name'] = 'email-address'
+  q['General Support'] = 'support'
+  q['Frontend Team'] = 'frontend'
+  q['Backend Team'] = 'backend'
+end
+
+default['osl-rt']['fqdn'] = 'support.osuosl.intnet'
```
Comment (ramereth) [inline documentation/resource_osl_request_tracker.md]: Please use `example.org` as the test domain globally.

File: kitchen.yml [hunk 5]
```diff
@@ -5,10 +5,16 @@ verifier:
 provisioner:
   name: chef_infra
   enforce_idempotency: true
+  log_level: 'info'
```
Comment (ramereth) [inline kitchen.yml]: Please remove this if you're not needing it anymore.

File: kitchen.yml [hunk 6]
```diff
@@ -5,10 +5,16 @@ verifier:
...
   multiple_converge: 2
   deprecations_as_errors: true
 
 suites:
   - name: default
     run_list:
-      - recipe[osl-rt::default]
+      - recipe[osl-rt-test::osl_request_tracker]
+    provisioner:
+      multiple_converge: 3
```
Comment (ramereth) [inline kitchen.yml]: Just remove this and update the global one.

File: kitchen.yml [hunk 7]
```diff
@@ -5,10 +5,16 @@ verifier:
...
+    driver:
+      name: openstack
+      flavor_ref: 'm1.large'
```
Comment (ramereth) [inline kitchen.yml]: This probably should also just be a global setting.

File: libraries/helpers.rb [hunk 8]
```diff
@@ -0,0 +1,129 @@
+module OslRT
+  module Cookbook
+    module Helpers
+      # Take in a hashmap containing the properties we'd like to set
+      # for the RT instance, and convert to the perl config standard
+      # Returns a string
+      def parse_config(hOptions)
```
Comment (ramereth) [inline libraries/helpers.rb:50]: Please prefix all of the public methods with `osl_rt_`. For example, this should be `osl_rt_parse_config`. Also, if none of these methods need to be used publicly (i.e in a recipe), then I'd recommend you move them into being private.

File: libraries/helpers.rb [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        strConfig = ''
+        hOptions.each do |key, value|
+          # If the key is _Plugins, this is a special case
+          if key == '_Plugins'
```
Comment (ramereth) [inline libraries/helpers.rb]: Could this be changed to a case statement perhaps?

File: libraries/helpers.rb [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+                       end
+        end
+        strConfig
+      end
+
+      # Take in an array of plugins to add
+      # and return a string to append to the config file
+      def parse_plugin(arrPlugins)
+        strConfig = ''
+        arrPlugins.each do |plugin|
```
Comment (ramereth) [inline libraries/helpers.rb]: Unless this causes an issue, sorting is usually good.
```suggestion
        arrPlugins.sort.each do |plugin|
```

File: recipes/default.rb [hunk 11]
```diff
@@ -15,3 +15,174 @@
...
+config_options['$DatabaseName'] = node['osl-rt']['db']['name']
+config_options['$DatabaseUser'] = node['osl-rt']['db']['username']
+config_options['$DatabasePassword'] = node['osl-rt']['db']['password']
+config_options['_Plugins'] = node['osl-rt']['plugins']
+config_options['_Lifecycles'] = node['osl-rt']['lifecycles']
+
+# Set up the queue emails
+rt_emails = init_emails(node['osl-rt']['queues'], node['osl-rt']['fqdn'], node['osl-rt']['default'])
+
+config_options['$RTAddressRegexp'] = "^(#{rt_emails.join('|')}(-comment)?\@(#{node['osl-rt']['fqdn']}))"
```
Comment (ramereth) [inline recipes/default.rb]: Any reason this can't be in a library method?

File: recipes/default.rb [hunk 12]
```diff
@@ -15,3 +15,174 @@
...
+
+# Set up the hostname as the website
+hostname node['hostname'] do
+  aliases ["#{node['osl-rt']['fqdn']}"]
+end
```
Comment (ramereth) [inline recipes/default.rb]: Why is this needed? We do this else where in production and I don't want this to conflict with that.

File: recipes/default.rb [hunk 13]
```diff
@@ -15,3 +15,174 @@
...
+
+# Parse the config hash to a string we can use as
+# the RT_SiteConfig.pm file.
+strConfigFile = parse_config(config_options)
```
Comment (ramereth) [inline recipes/default.rb]: Unless you use this variable more than once, I'd recommend you just use the method directly in the resource below.

File: recipes/default.rb [hunk 14]
```diff
@@ -15,3 +15,174 @@
...
+
+# Root Account
+template '/root/.rtrc' do
+  source 'rt/rtrc.erb'
+  cookbook 'osl-rt'
+  user 'root'
+  group 'root'
```
Comment (ramereth) [inline recipes/default.rb]: This is the default, please remove.

File: recipes/default.rb [hunk 15]
```diff
@@ -15,3 +15,174 @@
...
+
+# Add the RT command to the root user's PATH
+link '/usr/local/sbin/rt' do
+  to '/opt/rt/bin/rt'
+end
+
+# RT Initial Configuration.
+file '/opt/rt/etc/RT_SiteConfig.pm' do
+  content strConfigFile
+  user 'root'
```
Comment (ramereth) [inline recipes/default.rb]: This is the default, please remove.

File: recipes/default.rb [hunk 16]
```diff
@@ -15,3 +15,174 @@
...
+  group 'apache'
+  mode '0640'
+  sensitive true
+  notifies :reload, 'apache2_service[osuosl]'
+end
+
+# Initalize the DB
+execute 'init-db-rt' do
+  command <<-EOH
```
Comment (ramereth) [inline recipes/default.rb]: This properly removes the prepended spaces when executing. I usually like doing this if possible as it makes reading the output (later if needed) easier.
```suggestion
  command <<~EOH
```

File: recipes/default.rb [hunk 17]
```diff
@@ -15,3 +15,174 @@
...
+  include_template true
+  include_directory 'rt'
+  include_name 'rt'
+  include_params('domain': node['osl-rt']['fqdn'])
+end
+
+# Forcefully reload Apache during the initial run, in order to allow for setting up the queues properly.
+service 'httpd' do
+  action :reload
+  not_if { ::File.exist?('/var/spool/mail/nobody') }
```
Comment (ramereth) [inline recipes/default.rb]: Might want to add a comment for why this is needed.

File: recipes/default.rb [hunk 18]
```diff
@@ -15,3 +15,174 @@
...
+end
+
+# Set up the queues in RT
+node['osl-rt']['queues'].each do |pt, email|
+  execute "Creating RT queue for #{pt}" do
+    command <<~EOL
```
Comment (ramereth) [inline recipes/default.rb]: I like doing "end of command" or `EOC` for execute resources.
```suggestion
    command <<~EOC
```

File: recipes/default.rb (hunk 12 above)
Comment (zwall-bp) [reply recipes/default.rb]: In order to use the [RT CLI](https://rt-wiki.bestpractical.com/wiki/CLI) , allowing us to create queues via Chef, we need to set `server` in [`/root/.rtrc`](https://github.com/osuosl-cookbooks/osl-rt/blob/zwall/init/templates/default/rt/rtrc.erb) to the domain name of the website. Unfortunately, there isn't a config option for setting a `Host` HTTP header, to allow us to instead use `127.0.0.1`.

To prevent messing with `/etc/hosts`, I'll look into using the `HOSTALIASES` envar for the commands that make API calls to the website.

File: libraries/helpers.rb (hunk 8 above)
Comment (ramereth) [reply libraries/helpers.rb:50]: This still needs to be addressed.

**Review by ramereth — COMMENTED** [review]
Sorry, that was for a different PR.

File: libraries/helpers.rb (hunk 8 above)
Comment (zwall-bp) [reply libraries/helpers.rb:50]: The latest commit has classified all internal functions as `private`. The private declaration is [at the bottom of the module](https://github.com/osuosl-cookbooks/osl-rt/blob/zwall/init/libraries/helpers.rb#L149), but I'll update it to make it more salient for what is public and private.

Comment (ramereth) [reply libraries/helpers.rb:50]: Oh I missed that!

File: documentation/resource_osl_request_tracker.md [hunk 19]
```diff
@@ -0,0 +1,87 @@
...
+  },
+}
+
+# Example of additional config options
+rt_configs = {
+  '$WebPort' => '443',
+  '$SetOutgoingMailFrom' => 'support@osuosl.org'
+  '$LogoURL' -> 'https://osuosl.org/logos/osl.jpg'
+}
+```
```
Comment (ramereth) [inline documentation/resource_osl_request_tracker.md]: I think this needs to be updated as the resource no longer exists.

File: test/cookbooks/osl-rt-test/attributes/osl_request_teacker.rb [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+      '' => [ 'order in' ],
+      'order in' => [ 'order work' ],
+      'order work' => [ 'order up' ],
+    },
+
+    'rights' => {
+      '* -> order in' => 'ResetOrder',
+    },
+  },
+}
```
Comment (ramereth) [inline test/cookbooks/osl-rt-test/attributes/osl_request_teacker.rb]: Sane defaults should be set in the primary cookbook attribute file, not the test cookbook. Any overrides should happen in the recipe as `node.default`.

File: recipes/default.rb [hunk 21]
```diff
@@ -15,3 +15,166 @@
...
+template '/root/.rtrc' do
+  source 'rt/rtrc.erb'
+  cookbook 'osl-rt'
+  mode '0600'
+  sensitive true
+  variables(root_pass: node['osl-rt']['root-password'])
+end
+
+# Set up user for mail
+user node['osl-rt']['default'] do
```
Comment (ramereth) [inline recipes/default.rb]: I think this is a better name.
```suggestion
user node['osl-rt']['user'] do
```

File: recipes/default.rb [hunk 22]
```diff
@@ -15,3 +15,166 @@
...
+end
+
+# User defined Hostalias file in order to patch into the RT site with the RT CLI/procmail
+['root', "/home/#{node['osl-rt']['default']}"].each do |file_path|
+  file "/#{file_path}/.rthost" do
+    content <<~EOF
+      rtlocal localhost
+    EOF
+  end
+end
```
Comment (ramereth) [inline recipes/default.rb]: This is a better way to visualize it I think.
```suggestion
[
  '/root',
  "/home/#{node['osl-rt']['default']}"
].each do |file_path|
  file "#{file_path}/.rthost" do
    content <<~EOF
      rtlocal localhost
    EOF
  end
end
```

File: recipes/default.rb [hunk 23]
```diff
@@ -15,3 +15,166 @@
...
+  notifies :reload, 'apache2_service[osuosl]'
+end
+
+# Initalize the DB
+execute 'init-db-rt' do
+  command <<-EOC
+    /opt/rt/sbin/rt-setup-database \
+      --action init \
+      --dba #{node['osl-rt']['db']['username']} \
+      --dba-password #{node['osl-rt']['db']['password']} \
```
Comment (ramereth) [inline recipes/default.rb]: These should be pulled from a databag. Secrets should not be stored in attributes.

File: recipes/default.rb [hunk 24]
```diff
@@ -15,3 +15,166 @@
...
+
+# Set up web app
+apache_app node['osl-rt']['fqdn'] do
+  directory '/opt/rt/share/html'
+  include_config true
+  include_template true
+  include_directory 'rt'
+  include_name 'rt'
+  include_params('domain': node['osl-rt']['fqdn'])
+  server_aliases ['rtlocal']
```
Comment (ramereth) [inline recipes/default.rb]: This should be an attribute and be empty by default.

File: recipes/default.rb [hunk 25]
```diff
@@ -15,3 +15,166 @@
...
+  content osl_rt_init_config
+  group 'apache'
+  mode '0640'
+  sensitive true
+  notifies :reload, 'apache2_service[osuosl]'
+end
+
+# Initalize the DB
+execute 'init-db-rt' do
+  command <<-EOC
```
Comment (ramereth) [inline recipes/default.rb]: Please use `~` globally for these "heredoc" blocks of code. This automatically removes the prepended space when it runs the command and makes it easier to read the output if needed.
```suggestion
  command <<~EOC
```

File: recipes/default.rb [hunk 26]
```diff
@@ -15,3 +15,166 @@
...
+    creates "/tmp/#{email}done"
+  end
+end
+
+# Nobody mail directory
+file '/var/spool/mail/nobody' do
+  owner 'nobody'
+  group 'mail'
+  mode '0660'
+end
```
Comment (ramereth) [inline recipes/default.rb]: Why does this need to be created?

File: recipes/default.rb [hunk 27]
```diff
@@ -15,3 +15,166 @@
...
+include_recipe 'osl-mysql::client'
+include_recipe 'yum-osuosl'
+include_recipe 'perl'
+
+package %w(request-tracker mutt procmail)
+
+# Root Account
+template '/root/.rtrc' do
+  source 'rt/rtrc.erb'
+  cookbook 'osl-rt'
```
Comment (ramereth) [inline recipes/default.rb]: This isn't needed when we're doing it from a recipe. Please remove globally. We only need this inside of a resource.

File: test/cookbooks/osl-rt-test/recipes/osl_request_tracker.rb [hunk 28]
```diff
@@ -0,0 +1,28 @@
...
+  username node['osl-rt']['db']['username']
+  password node['osl-rt']['db']['password']
+end
+
+# Request Tracker
+include_recipe 'osl-rt'
+
+# Wait a bit so the RT instance could start back up
+chef_sleep 'Wait a bit' do
+  seconds 5
```
Comment (ramereth) [inline test/cookbooks/osl-rt-test/recipes/osl_request_tracker.rb]: Just do this to keep it cleaner.
```suggestion
chef_sleep '5' do
```

File: test/integration/default/default_test.rb [hunk 29]
```diff
@@ -0,0 +1,190 @@
...
+    'backend-comment@example.org local:$myhostname',
+    'backend@example.org local:$myhostname',
+    'devops-comment@example.org local:$myhostname',
+    'devops@example.org local:$myhostname',
+    'advertising-comment@example.org local:$myhostname',
+    'advertising@example.org local:$myhostname',
+    'board-comment@example.org local:$myhostname',
+    'board@example.org local:$myhostname',
+    'support-comment@example.org local:$myhostname',
+    'support@example.org local:$myhostname',
```
Comment (ramereth) [inline test/integration/default/default_test.rb]: Please sort this alphabetically.

**Review by ramereth — CHANGES_REQUESTED** [review]
Looks pretty good, but I have a few suggested changes that might make this better.

File: recipes/default.rb (hunk 26 above)
Comment (zwall-bp) [reply recipes/default.rb]: This was a leftover from the `osl-admin` transition. It's original goal was to be the end of the line for the RT mail that had no recipient. But that has since been changed to be the base user of the VM (`almalinux`, `centos`).

File: recipes/default.rb (hunk 27 above)
Comment (zwall-bp) [reply recipes/default.rb]: Unfortunately, it would be preferable to have the RT config set up, as we do run RT commands that make calls to the API. This is done around line `134`, with setting up the queues.

Unless, are you saying that we should instead have the config options in the command itself?

File: recipes/default.rb (hunk 26 above)
Comment (ramereth) [reply recipes/default.rb]: What did we have it set in osl-admin?

File: recipes/default.rb (hunk 27 above)
Comment (ramereth) [reply recipes/default.rb]: I'm preferring to removing the `cookbook` parameters. Sorry for the confusion.

File: libraries/helpers.rb [hunk 30]
```diff
@@ -3,23 +3,25 @@ module Cookbook
     module Helpers
       # Initalize the configuration options given the attributes
       def osl_rt_init_config
+        rt_secrets = data_bag_item(*node['osl-rt']['data-bag'])
```
Comment (ramereth) [inline libraries/helpers.rb]: This attribute doesn't have a default setting at all in the default attributes file. Also I prefer to have the same databag name but the item can be attribute. For this I'd prefer to have the databag name be `request-tracker` and the default item name could be `default`.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: test/cookbooks/osl-rt-test/attributes/osl_request_teacker.rb (hunk 20 above)
Comment (ramereth) [reply test/cookbooks/osl-rt-test/attributes/osl_request_teacker.rb]: @zwall-bp I believe this still needs to be addressed.

File: documentation/resource_default.md [hunk 31]
```diff
@@ -0,0 +1,76 @@
...
+      'order work' => [ 'order delayed', 'order up' ],
+      'order delayed' => [ 'order work' ],
+    },
+
+    'rights' => {
+      '* => order in' => 'ResetOrder',
+    }
+  }
+}
+```
```
Comment (ramereth) [inline documentation/resource_default.md]: Let's move this into the main README.md for now as it's a little confusing that it's here and also hard to find/notice.

File: spec/unit/recipes/default_spec.rb [hunk 32]
```diff
@@ -19,14 +19,192 @@
...
+    osl-apache osl-apache::mod_ssl osl-apache::mod_perl
+    osl-mysql::client yum-osuosl perl
+    osl-postfix::server postfix::aliases
+    postfix::access postfix::transports
+  ).each do |r|
+    it { is_expected.to include_recipe(r) }
+  end
+
+  # RT Site Config
+  it {
```
Comment (ramereth) [inline spec/unit/recipes/default_spec.rb]: For multiline tests, please do these in `it do` blocks instead.

File: templates/default/rt/rtrc.erb [hunk 33]
```diff
@@ -0,0 +1,3 @@
+server http://rtlocal
```
Comment (ramereth) [inline templates/default/rt/rtrc.erb]: Does this need to be different depending on how this gets deployed in production?

File: test/cookbooks/osl-rt-test/attributes/osl_request_teacker.rb (hunk 20 above)
Comment (ramereth) [reply test/cookbooks/osl-rt-test/attributes/osl_request_teacker.rb]: Please move all of these into the test recipe as `node.default` which is how we would normally do this in production.

File: test/cookbooks/osl-rt-test/files/default/rt/rt.conf [hunk 34]
```diff
@@ -0,0 +1,17 @@
+RewriteEngine On
+RewriteRule ^/([0-9]+)$ https://request.osuosl.intnet/Ticket/Display.html?id=$1 [QSA,L]
```
Comment (ramereth) [inline test/cookbooks/osl-rt-test/files/default/rt/rt.conf]: I prefer to use this domain for testing.
```suggestion
RewriteRule ^/([0-9]+)$ https://request.example.org/Ticket/Display.html?id=$1 [QSA,L]
```

File: templates/default/rt/rtrc.erb (hunk 33 above)
Comment (zwall-bp) [reply templates/default/rt/rtrc.erb]: This can have an issue if we are wanting to deploy multiple RT instances on the same server. Since this was last implemented, the "domain" tied to the site has changed to an attribute. So, it should theoretically allow for multiple RT sites on a single VM. (Once we turn this into a resource)

Unfortunately, having a secondary domain name has to be done as we need a domain name to access the site from the VM. But, modifying `/etc/hosts` would be too intrusive. Instead, we are modifying the available hosts through the `HOSTALIASES` envar.

Doing HTTP requests to `127.0.0.1` with `Host` in the header will not work.

Comment (ramereth) [reply templates/default/rt/rtrc.erb]: I think it's safe to assume there will only be one instance of RT per system.

**Comment by zwall-bp** [comment]
The attributes are now set using databags.
 
I'd like to point out that attributes are still present in the codebase. Since the attributes should be easily accessible to give other OSL developers an example for how they are set. Along with this, it allows us to set defaults for everything, then have the data bag entries take precedence. As choosing a specific data bag would not allow us to set defaults.

File: libraries/helpers.rb [hunk 35]
```diff
@@ -1,6 +1,16 @@
 module OslRT
   module Cookbook
     module Helpers
+      # Take all of the config settings set in the databag, and overwrite the default attributes.
+      def osl_rt_set_attributes
+        rt_secrets = data_bag_item('request-tracker', node['osl-rt']['data-bag'])
+
+        # Iterate over all keys which will be replaced
+        %w(fqdn user internal-comain db queues plugins lifecycles).each do |key|
+          node.override['osl-rt'][key] = rt_secrets[key] if rt_secrets[key]
```
Comment (ramereth) [inline libraries/helpers.rb]: We shouldn't be using attributes anymore, just do it all from the databag.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: README.md [hunk 36]
```diff
@@ -1,21 +1,105 @@
...
+`db.host`        | String | The hostname of the DB server                                | `localhost`
+`db.name`        | String | The DB name on the DB server                                 | `rt`
+`queues`         | Hash   | The queues and emails available for RT. The key is the pretty print, and the value is a email-valid name. | `{'Support Example': nil}` Any nil-valued key will be ignored.
+`plugins`        | Array  | A list of [plugins](https://rt-wiki.bestpractical.com/wiki/Extensions) to add to the RT site | `[]`
+`lifecycles`     | Hash   | `WARNING: Default set to a custom lifecycle, explicitly set this to nil for actual default.` Any [custom lifecycles](https://docs.bestpractical.com/rt/4.4.1/customizing/lifecycles.html) to make available in RT | A custom example of a default lifecycle.
+
+### Example Data Bag Attributes
+
+```json
+# Create three different Queues. The keys are the "pretty print" name, while the values are the email name.
```
Comment (ramereth) [inline README.md]: JSON doesn't have comments and this makes it hard to read. Can you please just show the whole databag as an example?

File: test/cookbooks/osl-rt-test/recipes/osl_request_tracker.rb [hunk 37]
```diff
@@ -0,0 +1,28 @@
...
+# Database
+osl_mysql_test 'rt' do
+  username 'rt-user'
+  password 'rt-password'
+end
+
+# Request Tracker
+include_recipe 'osl-rt'
+
+# Restart Apache
```
Comment (ramereth) [inline test/cookbooks/osl-rt-test/recipes/osl_request_tracker.rb]: Please add a comment for why this is needed.

**Review by ramereth — APPROVED** [review]
I think this is finally ready. Please do a major bump for the first release. You will have to manually make a PR adding it to the production environment.

**Comment by zwall-bp** [comment]
!bump major

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `main` and has automatically performed a major-level version bump to v1.0.0.  Have a nice day!
